cmd/vaults/commands.go

```diff
@@ -788,139 +775,143 @@ func parseVaultName(name string) (ns string, rel string, err error) {
 	return
 }
 
-func inspectTable(ctx context.Context, tx pgx.Tx, rel string) ([]app.Column, error) {
-	rows, err := tx.Query(ctx,
-		`
-		WITH primary_key_info AS
-			(SELECT tc.constraint_schema,
-					tc.table_name,
-					ccu.column_name
-			FROM information_schema.table_constraints tc
-			JOIN information_schema.constraint_column_usage AS ccu USING (CONSTRAINT_SCHEMA, CONSTRAINT_NAME)
-			WHERE constraint_type = 'PRIMARY KEY' ),
-			array_type_info AS
-				(SELECT c.table_name,
-						c.column_name,
-						pg_catalog.format_type(t.oid, NULL) AS full_data_type
-				FROM information_schema.columns AS c
-				JOIN pg_catalog.pg_type AS t ON c.udt_name = t.typname
-				WHERE c.data_type = 'ARRAY')
-		SELECT
-			c.column_name,
-			CASE
-			WHEN c.data_type = 'ARRAY' THEN ati.full_data_type
-			ELSE c.data_type
-			END AS data_type,
-			c.is_nullable = 'YES' AS is_nullable,
-			pki.column_name IS NOT NULL AS is_primary
-		FROM information_schema.columns AS c
-		LEFT JOIN primary_key_info pki ON c.table_schema = pki.constraint_schema
-			AND pki.table_name = c.table_name
-			AND pki.column_name = c.column_name
-		LEFT JOIN array_type_info ati ON c.table_name = ati.table_name
-    		AND c.column_name = ati.column_name
-			WHERE c.table_name = $1;
-		`, rel,
-	)
-	if err != nil {
-		return []app.Column{}, fmt.Errorf("failed to fetch schema")
+func validateBeforeAndAfter(before, after, at string) (app.Timestamp, app.Timestamp, error) {
+	if !strings.EqualFold(at, "") {
+		before, after = at, at
 	}
-	defer rows.Close()
-
-	var colName, typ string
-	var isNull, isPrimary bool
-	var columns []app.Column
-	for rows.Next() {
-		if err := rows.Scan(&colName, &typ, &isNull, &isPrimary); err != nil {
-			return []app.Column{}, fmt.Errorf("scan: %s", err)
-		}
 
-		columns = append(columns, app.Column{
-			Name:      colName,
-			Typ:       typ,
-			IsNull:    isNull,
-			IsPrimary: isPrimary,
-		})
+	b, err := app.ParseTimestamp(before)
+	if err != nil {
+		return app.Timestamp{}, app.Timestamp{}, err
 	}
-	return columns, nil
-}
 
-func createVault(
-	ctx context.Context,
-	dburi string,
-	ns string,
-	rel string,
-	provider string,
-	account *app.Account,
-	cacheDuration int64,
-) (exists bool, err error) {
-	bp := vaultsprovider.New(provider)
-	req := app.CreateVaultParams{
-		Account:       account,
-		Vault:         app.Vault(fmt.Sprintf("%s.%s", ns, rel)),
-		CacheDuration: app.CacheDuration(cacheDuration),
+	a, err := app.ParseTimestamp(after)
+	if err != nil {
+		return app.Timestamp{}, app.Timestamp{}, err
 	}
 
-	if dburi == "" {
-		if err := bp.CreateVault(ctx, req); err != nil {
-			return false, fmt.Errorf("create vault: %s", err)
-		}
+	return b, a, nil
+}
 
-		return exists, nil
-	}
+// DatabaseStreamSetup setups the database for streaming.
+type DatabaseStreamSetup struct {
```

Comment:
created this object to make it easier to setup the database for replication. it can:
- create the publication in Postgres
- fetch the schemas of the tables in Postgres so that we can create a similar table in Duckdb